Bookings: Add sort options

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

Closes WOOMOB-1238

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR implements the sorting feature for the booking list:
- Adds an action sheet for sorting options.
- Sends selected order to booking list and search view models.
- Updates BookingsRemote, BookingAction and BookingStore to support sending order to the remote.
- Updates the view models to reload when the order changes.

Note: Upon change of order, I reload the lists disregard of whether the list is empty or not - to avoid edge cases. Let me know if you think it makes more sense to not reload when the list is empty.

## Testing steps
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

1. Log in to a CIAB store with existing bookings.
2. In the booking list, select Sort by. Confirm that the action sheet matches the design.
3. Select an option and confirm that both the list and the search view are updated for all tabs. 

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

- Tested with simulator iPhone 17.
- Added unit tests for the view models.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->


https://github.com/user-attachments/assets/39038035-083e-4e12-b12c-84347e18a6e1



---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Add sheet for sorting options

itsmeichigo committed:
Move sortBy to view model as a published property

itsmeichigo committed:
Apply new order to the sort descriptor

itsmeichigo committed:
Add sorting option for fetching bookings

itsmeichigo committed:
Update booking fetch when reordering

itsmeichigo committed:
Update sort order for booking search

itsmeichigo committed:
Merge branch 'woomob-1393-booking-list-search' into woomob-1238-ios-booking-list-sorting

itsmeichigo committed:
Add tests for updateSortOrder

itsmeichigo committed:
Remove duplicates of order

adborbas commented:
Hey @itsmeichigo, the code looks good and works as described. I know that things are still in WIP, but I have noticed that the sort-by selector differs between the Products and Bookings tab.
- The header "Sort by" has a bigger spacing on Products, and per design, it seems like it should.
- The case is different as well. Products: "Newest to Oldest" on Bookings: "Newest to oldest"

| Type | Products | Bookings |
| --- | --- | --- | 
| iPhone | <img width="590" height="1278" alt="Products" src="https://github.com/user-attachments/assets/a7cde2eb-bdd2-46d5-afc5-090449252581" /> | <img width="590" height="1278" alt="Products 2" src="https://github.com/user-attachments/assets/696ba504-6528-4f80-9aec-9baf21129c78" /> | 
| iPad | <img width="1194" height="834" alt="IMG_0017" src="https://github.com/user-attachments/assets/45bb1f12-7ef4-4bb8-b1d8-9bad58ac2157" /> | <img width="1194" height="834" alt="IMG_0016" src="https://github.com/user-attachments/assets/d0a6153d-f38a-47ba-ab91-c7736f294a5c" /> |

itsmeichigo commented:
Thanks @adborbas - I left a question on Slack [p1760365242096829-slack-C09FHQNQERG] for our designer regarding your concerns, hopefully we'll get a response soon.

itsmeichigo committed:
Merge branch 'trunk' into woomob-1238-ios-booking-list-sorting

wagveloso commented:
Hi team. Yes, please you can use the products sort by UI as reference. I've also updated the Figma.
https://www.figma.com/design/lxixW6aBBNkCd9ooBK70Z1/CIAB-Mobile-Bookings?node-id=359-5652&t=BSAWUf38crWBTYP1-1

itsmeichigo committed:
Merge branch 'trunk' into woomob-1238-ios-booking-list-sorting

itsmeichigo committed:
Update padding and title for filter options

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/257">23.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

itsmeichigo committed:
Unify top padding for sorting options

itsmeichigo commented:
@adborbas I fixed the padding issue and updated the titles to stay consistent with Products tab. This PR is ready for another look.

<img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-15 at 16 41 43" src="https://github.com/user-attachments/assets/431e0cf4-c01e-42e2-aadb-04b331bc1f4f" />

adborbas commented:
Hi @itsmeichigo, Nice! I have one thing to note, but I'm okay with addressing this in another PR.

There is no visual clue that we are loading data, and the list seems to be rearranged out of nowhere. You can make this obvious by throttling the network on iOS and changing the sort order on `All`. 

![ScreenRecording_10-15-2025 12-23-20_1](https://github.com/user-attachments/assets/dc00edb1-c6bc-4a79-93b5-a34642ae1397)
